src/pubkeyshelper.ts

```diff
@@ -0,0 +1,80 @@
+import { AxiosResponse } from 'axios'
+
+import { defaultHttpRequestHandler } from './defaulthttprequesthandler'
+import { PublicKey } from './index'
+import * as jwkUtils from './jwkutils'
+
+interface Options {
+  expiresSkew?: number
+  defaultAlgorithms?: string[]
+}
+
+export class PubkeysHelper {
+  private requestHandler: (
+    method: string,
+    url: string,
+    headers?: Record<string, string | number>,
+    body?: unknown
+  ) => Promise<AxiosResponse>
+
+  public constructor(
+    httpRequestHandler?: (
+      method: string,
+      url: string,
+      headers?: Record<string, string | number>,
+      body?: unknown
+    ) => Promise<AxiosResponse>
+  ) {
+    this.requestHandler = httpRequestHandler || defaultHttpRequestHandler
+  }
+
+  public async fetchJwkKeys(url: string, options: Options = {}): Promise<Record<string, PublicKey> | null> {
+    const defaultAlgorithms = options.defaultAlgorithms || []
+    delete options.defaultAlgorithms
+
+    let updatedOptions: Record<string, number> = {}
+    if (options.expiresSkew != undefined) {
+      updatedOptions = { expiresSkew: options.expiresSkew }
+    }
+
+    const result = await this.requestHandler('GET', url, {}, null)
+    if (!result) {
+      return null
+    }
+
+    const pubKeys = this.formatPublicKeys(result.data, url, defaultAlgorithms, updatedOptions)
+    if (!pubKeys) {
+      return null
+    }
+    return pubKeys
+  }
+
+  private formatPublicKeys(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    response: any,
```

Comment:
Can we make a type for this? It just seems to be getting the jwkUris passed which seems to look like this:
```
$ curl https://www.googleapis.com/oauth2/v3/certs                                                                                                                                                                                                           [13:13]
{
  "keys": [
    {
      "kid": "fd48a75138d9d48f0aa635ef569c4e196f7ae8d6",
      "kty": "RSA",
      "use": "sig",
      "alg": "RS256",
      "e": "AQAB",
      "n": "8KImylelEspnZ0X-ekZb9VPbUFhgB_yEPJuLKOhXOWJLVsU0hJP6B_mQOfVk0CHm66UsAhqV8qrINk-RXgwVaaFLMA827pbOOBhyvHsThcyo7AY5s6M7qbftFKKnkfVHO6c9TsQ9wpIfmhCVL3QgTlqlgFQWcNsY-qemSKpqvVi-We9I3kPvbTf0PKJ_rWA7GQQnU_GA5JRU46uvw4I1ODf0icNBHw7pWc7oTvmSl1G8OWABEyiFakcUG2Xd4qZnmWaKwLHBvifPuIyy2vK-yHH91mVZCuleVu53Vzj77RgUtF2EEuB-zizwC-fzaBmvnfx1kgQLsdK22J0Ivgu4Xw"
    },
    {
      "kty": "RSA",
      "e": "AQAB",
      "use": "sig",
      "kid": "911e39e27928ae9f1e9d1e21646de92d19351b44",
      "alg": "RS256",
      "n": "4kGxcWQdTW43aszLmftsGswmwDDKdfcse-lKeT_zjZTB2KGw9E6LVY6IThJVxzYF6mcyU-Z5_jDAW_yi7D_gXep2rxchZvoFayXynbhxyfjK6RtJ6_k30j-WpsXCSAiNAkupYHUyDIBNocvUcrDJsC3U65l8jl1I3nW98X6d-IlAfEb2In2f0fR6d-_lhIQZjXLupjymJduPjjA8oXCUZ9bfAYPhGYj3ZELUHkAyDpZNrnSi8hFVMSUSnorAt9F7cKMUJDM4-Uopzaqcl_f-HxeKvxN7NjiLSiIYaHdgtTpCEuNvsch6q6JTsllJNr3c__BxrG4UMlJ3_KsPxbcvXw"
    }
  ]
}
```